doc/examples/parallel_planning/parallel_planning_tutorial.rst

```diff
@@ -0,0 +1,162 @@
+        max_acceleration_scaling_factor: 1.0
+        planning_time: 1.5
+
+Optionally, it is possible to define a custom stopping criterion and/or solution selection function. If none are passed as an argument to the :code:`plan(...)`,
+all pipelines use their complete planning time budget and afterwards the shortest path is chosen.
+
+For this example we're using the default stopping criterion and a solution selection criterion that choses the shortest solution:
+
+.. code-block:: c++
+
+    planning_interface::MotionPlanResponse getShortestSolution(const std::vector<planning_interface::MotionPlanResponse>& solutions)
+    {
+      // Find trajectory with minimal path
+      auto const shortest_solution = std::min_element(solutions.begin(), solutions.end(),
+        [](const planning_interface::MotionPlanResponse& solution_a,
+           const planning_interface::MotionPlanResponse& solution_b) {
+          // If both solutions were successful, check which path is shorter
+          if (solution_a && solution_b)
+          {
+            return robot_trajectory::path_length(*solution_a.trajectory_) <
+                   robot_trajectory::path_length(*solution_b.trajectory_);
+          }
+          // If only solution a is successful, return a
+          else if (solution_a)
+          {
+            return true;
+          }
+          // Else return solution b, either because it is successful or not
+          return false;
+        });
+      return *shortest_solution;
+    }
+
+Here is an example for a custom stopping criterion:
+
+.. code-block:: c++
+
+    // Stop parallel planning as soon as RRTConnect finds a solution
+    bool stoppingCriterion(
+        moveit_cpp::PlanningComponent::PlanSolutions const& plan_solutions,
+        moveit_cpp::PlanningComponent::MultiPipelinePlanRequestParameters const& plan_request_parameters)
+    {
+      // Read solutions that are found up to this point from a thread safe storage
+      auto const& solutions = plan_solutions.getSolutions();
+
+      // Stop parallel planning if the pipeline using RRTConnect finds a solution
+      for (auto const& solution : solutions)
+      {
+          if (solution.planner_id_ == "RRTConnectkConfigDefault")
+          {
+            // Return true to abort the other pipelines
+            return true;
+          }
+      }
+      // Return false when parallel planning should continue
+      return false;
+    }
+
+Once :code:`MultiPipelinePlanRequestParameters` and optionally :code:`SolutionCallbackFunction` and/or :code:`StoppingCriterionFunction` are defined, we call :code:`plan(...)`:
+
+.. code-block:: c++
+
+    auto plan_solution = planning_component_->plan(multi_pipeline_plan_request, &getShortestSolution);
+
+Tips
+----
+
+- When you want to use multiple pipelines with the same planner parallel it is recommended to initialize multiple planning pipelines in moveit_cpp rather than using the same one in multiple parallel planning requests
```

Comment:
Is this what you are saying here?:

>If you want to use multiple planners from the same planning pipeline (e.g., PRM and RRT from the OMPL planning pipeline), then it is recommended to use multiple instances of that planning pipeline.

Why "recommended"? What will break if you don't? Why is it not required?

Reply:
Yes, I rephrased the sentence to:
```
When you want to use different planners of the same pipeline (e.g. PILZ planner with PTP and LIN) in parallel, it is recommended to initialize multiple planning pipelines in moveit_cpp rather than using the same one in multiple parallel planning requests
```
I'd rather say recommend than required since it can be possible that a planner plugin can serve multiple requests at the same time (it seems to work with OMPL) but depending on the planner it could also result in undefined behavior, race conditions or "hidden sequential planning". Using two different pipelines in this case is the safest option for now but it would be nice to investigate the parallelism/concurrency capabilities of our plugins more in the future

Reply:
I've added a second OMPL pipeline that uses RRT*